test/test_amqp.py

```diff
@@ -49,3 +54,147 @@ def test_other_error(uris):
     amqp_uri = uris['bad_port']
     with pytest.raises(socket.error):
         verify_amqp_uri(amqp_uri)
+
+
+def test_get_connection(rabbit_config):
+    amqp_uri = rabbit_config['AMQP_URI']
+    connection_ids = []
+
+    with get_connection(amqp_uri) as connection:
+        connection_ids.append(id(connection))
+        assert isinstance(connection, Connection)
+
+    with get_connection(amqp_uri) as connection:
+        connection_ids.append(id(connection))
+        assert len(set(connection_ids)) == 1
+
+
+class TestGetProducer(object):
+
+    @pytest.fixture(params=[True, False])
+    def confirms(self, request):
+        return request.param
+
+    def test_get_producer(self, rabbit_config, confirms):
+        amqp_uri = rabbit_config['AMQP_URI']
+        producer_ids = []
+
+        with get_producer(amqp_uri, confirms) as producer:
+            producer_ids.append(id(producer))
+            transport_options = producer.connection.transport_options
+            assert isinstance(producer, Producer)
+            assert transport_options['confirm_publish'] is confirms
+
+        with get_producer(amqp_uri, confirms) as producer:
+            producer_ids.append(id(producer))
+            assert len(set(producer_ids)) == 1
+
```

Comment:
This passes:

``` python
    def test_different_producers(self, rabbit_config):
        amqp_uri = rabbit_config['AMQP_URI']
        producer_ids = []

        # get a producer
        with get_producer(amqp_uri, True) as confirmed_producer:
            producer_ids.append(id(confirmed_producer))
            assert len(set(producer_ids)) == 1

        # get a producer with the same parameters
        with get_producer(amqp_uri, True) as confirmed_producer:
            producer_ids.append(id(confirmed_producer))
            assert len(set(producer_ids)) == 1

        # get a producer with different parameters
        with get_producer(amqp_uri, False) as unconfirmed_producer:
            producer_ids.append(id(unconfirmed_producer))
            assert len(set(producer_ids)) == 2
```

But I'm not sure it's a valid test. Publish confirms are a parameter of the connection -- isn't it internal to kombu whether you get the same or different producer instances back in this case?